server/code_generator.py

```diff
@@ -0,0 +1,34 @@
+import random
+import string
+
+bad_words_file = open('server/bad_words.txt', 'r')
+BAD_WORDS_LIST = bad_words_file.read().split('\n')
+bad_words_file.close()
+
+def generate_code(num_of_chars):
+    code = ''
+
+    while not is_valid(code):
+        code = generate_random_elems(num_of_chars)
+
+    return code
+
+def generate_random_elems(num_of_chars):
+    POSSIBLE_CHARS = string.ascii_letters + string.digits
+    code = ''
+
+    for x in range(num_of_chars):
+        random_elem = random.choice(POSSIBLE_CHARS)
+        code += random_elem
+
+    return code
+
+def is_valid(code):
+    if not "".join(code.split()):
```

Comment:
Not sure what you're trying to accomplish here... the code shouldn't have spaces, right? So why are we splitting?

Reply:
i was using this as a way to strip white space so that empty strings and codes with blank spaces are marked as invalid. The if statement checks if the string is empty, but it's not marked as empty if there is whitespace.

Reply:
Hm, i guess the code shouldn't have any spaces, but I think that is_valid should not return true for a string that only contains whitespace. That's one of the tests I wrote for it

Reply:
The codes that you're generating can only have the characters A-Z, a-z, and 0-9, so whitespace shouldn't be a problem